src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1420,20 +1420,27 @@ private object ConvertValue(object value, _SqlMetaData metadata, bool isNull, re
                             sqlValue = new SqlDecimal((Decimal)value);
                         }
 
+                        SqlDecimal sqlValueBeforeAdjust = sqlValue;
```

Comment:
`sqlValue` could be changed by this block:
```
if (sqlValue.Scale != metadata.scale)
{
	sqlValue = TdsParser.AdjustSqlDecimalScale(sqlValue, metadata.scale);
}
```
We need original value before adjusting precision and scale for error message.